### DISCUSSION TITLE
Progressbar in core

### DISCUSSION THREAD

**Discussion by vindevoy** [posit-dev/py-shiny#1422] [Q&A] [answered]
Hi,

The example for the ProgressBar exists only in the express version.  I can't get it to work in the core version.  However, I see there is an express method in core that seems to render with the "with" context manager.  But also without example.

Would it be possible that an example is created to use the PB in core using this ?

**Comment by cpsievert** [comment, marked as answer]
The `ui.Progress` class should work both in Core and Express (the context manager approach should work in either mode as well). Here is an example if the `with` usage in Core:

https://shiny.posit.co/py/api/core/ui.Progress.html

**Comment by vindevoy** [comment]
My excuses if I overlooked this one in the core reference
